lsp-mode.el

```diff
@@ -3001,6 +3002,7 @@ If NO-MERGE is non-nil, don't merge the results but return alist workspace->resu
                               nil
                               target-workspaces))
              (error-callback (lambda (error)
+                               (funcall callback nil)
```

Comment:
nil is valid return value so you cannot use it to indicate an error: use something like `:error`.